src/net.cpp

```diff
@@ -1946,9 +1958,34 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
             // well for sanity.)
             bool block_relay_only = nOutboundBlockRelay < m_max_outbound_block_relay && !fFeeler && nOutboundFullRelay >= m_max_outbound_full_relay;
 
+            if (anchor) {
+                block_relay_only = true;
+                LogPrintf("Trying to make an anchor connection to %s\n", addrConnect.ToString());
+            }
+
             OpenNetworkConnection(addrConnect, (int)setConnected.size() >= std::min(nMaxConnections - 1, 2), &grant, nullptr, false, fFeeler, false, block_relay_only);
+
+            if (anchor) {
+                const auto v = GetCurrentAnchors();
+                if (std::find(v.begin(), v.end(), addrConnect) == v.end()) {
+                    LogPrintf("The anchor connection to %s FAILED\n", addrConnect.ToString());
+                }
+            }
+        }
+    }
+}
+
+std::vector<CAddress> CConnman::GetCurrentAnchors() const
+{
+    std::vector<CAddress> ret;
+    LOCK(cs_vNodes);
+    for (const CNode* pnode : vNodes) {
+        if (!pnode->fInbound && !pnode->m_manual_connection && !pnode->m_tx_relay) {
+            ret.push_back(pnode->addr);
         }
     }
+
+    return ret;
```

Comment:
2 questions about this:

1. Is there a possible race condition where `GetCurrentAnchors()` is called before the connection to the anchor node is complete? Or before the handshake is complete? Like is it possible that the test succeeds, no message is printed to the log, but then the connection still fails due to the peer not responding, etc.?

2. (and maybe this answers my first question) is there a better way of determining connection success than locking `cs_vNodes` and searching for the addr? Like, is there any sense in modifying `OpenNetworkConnection()` to return a success value?

Reply:
The `CConnman::OpenNetworkConnection()` call is synchronous, therefore there is no race with the following `CConnman::GetCurrentAnchors()` call, and I couldn't see a reason to modify it.

Reply:
What do you think about (2)? Could you get what you need if `OpenNetworkConnection()` returned `true` on success?

Reply:
`CConnman::OpenNetworkConnection()` has 5 call sites. What benefits of the `bool` return type for all callers? Unsuccessful connection is not an error.

And it seems orthogonal to this PR :)

Reply:
> ... is there a better way of determining connection success than locking `cs_vNodes` and searching for the addr?

The vector being searching in is very small :)

Reply:
gotcha, thanks!